helpers/misc.go

```diff
@@ -146,3 +147,28 @@ func MergeNonZero[T any](original T, overrides T) T {
 	}
 	return originalValue.Elem().Interface().(T)
 }
+
+// MergePathAndValueIntoMap takes a path in dot notation as a string and a value (also as a string for simplicity),
+// then merges this into the provided map. The value can be any type.
+func MergePathAndValueIntoMap(m map[string]any, path string, value any) error {
```

Comment:
The function signature may need to be changed depending on whether Zarf chart variables will have a `value:` field that is included the merge behavior. If a `value:` field is needed, then that may need to be added as an additional argument to this function.

https://github.com/defenseunicorns/zarf/pull/2403/files#r1543588026
https://github.com/defenseunicorns/zarf/pull/2403/files#r1543612278

Reply:
a `value:` field is not needed: https://github.com/defenseunicorns/zarf/pull/2403#discussion_r1543715691